app/serializers/search_form_serializer.rb

```diff
@@ -4,57 +4,79 @@ class SearchFormSerializer < ActiveModel::Serializer
   attributes :sort, :query
 
   def sort
-    {
-      '_script': {
-        'script': types_of_advice_sorting_expression,
-        'type': 'number',
-        'order': 'desc'
-      },
-      '_geo_distance': {
-        'advisers.location': object.coordinates.reverse,
-        'order': 'asc',
-        'unit': 'miles'
-      }
-    }.tap do |expression|
-      expression.delete(:_script) unless types_of_advice?
+    [].tap do |options|
+      if object.phone_or_online?
+        options << 'registered_name'
+      end
+
+      if types_of_advice?
+        options << {
+          _script: {
+            script: types_of_advice_sorting_expression,
+            type: 'number',
+            order: 'desc'
+          }
+        }
+      end
+
+      if object.face_to_face?
+        options << {
+          _geo_distance: {
+            'advisers.location': object.coordinates.reverse,
+            'order': 'asc',
+            'unit': 'miles'
+          }
+        }
+      end
     end
   end
 
   def query
     {
-      'filtered': {
-        'filter': {
-          'script': {
-            'script': types_of_advice_filter_expression
+      filtered: {
+        filter: {
+          bool: {
+            must: build_filters
           }
         },
-        'query': {
-          'bool': {
-            'must': (build_postcode_filters + build_investment_sizes_filter)
+        query: {
+          bool: {
+            must: build_queries
           }
         }
       }
-    }.tap do |expression|
-      expression[:filtered].delete(:filter) unless types_of_advice?
-    end
+    }
   end
 
   private
 
-  def build_postcode_filters
+  def build_filters
+    [].tap do |filters|
+      filters << { script: { script: types_of_advice_filter_expression } } if types_of_advice?
+      filters << { in: { other_advice_methods: object.remote_advice_method_ids } } if object.phone_or_online?
+    end
+  end
+
+  def build_queries
+    investment_size_queries.tap do |expression|
+      expression.push(*postcode_queries) if object.face_to_face?
+    end
+  end
+
+  def postcode_queries
     [
       {
-        'match': {
-          'postcode_searchable': true
+        match: {
+          postcode_searchable: true
         }
       },
       {
-        'nested': {
-          'path': 'advisers',
-          'filter': {
-            'geo_distance': {
-              'distance': '650miles',
-              'location': object.coordinates.reverse
+        nested: {
+          path: 'advisers',
+          filter: {
+            geo_distance: {
+              distance: '650miles',
```

Comment:
This literal (I mean '650' part) is a bit specific - it is a part of domain knowledge, rather than just an implementation detail. I think it might be a good idea to extract it at some point into a constant in `rad-core` project.

However, it's outside of PR scope.